src/main/java/com/google/api/codegen/transformer/ruby/RubySurfaceNamer.java

```diff
@@ -47,12 +52,18 @@ public RubySurfaceNamer(String packageName) {
         packageName);
   }
 
-  @Override
   /** The name of the class that implements snippets for a particular proto interface. */
+  @Override
   public String getApiSnippetsClassName(Interface interfaze) {
     return publicClassName(Name.upperCamel(interfaze.getSimpleName(), "ClientSnippets"));
   }
 
+  @Override
+  public String getPathTemplateName(
+      Interface service, SingleResourceNameConfig resourceNameConfig) {
+    return Name.from(resourceNameConfig.getEntityName(), "path", "template").toUpperUnderscore();
```

Comment:
It's used for a constant variable name. The default implementation produces stuff like `ShelfPathTemplate` for Ruby when we need `SHELF_PATH_TEMPLATE`.